Upstream: Is going to mark it an expected failure.
Source: philantrop@exherbo.org
Reason: Test is broken.

diff -Naur bzr-1.10.orig/bzrlib/tests/test_diff.py bzr-1.10/bzrlib/tests/test_diff.py
--- bzr-1.10.orig/bzrlib/tests/test_diff.py    2008-12-05 18:54:06.000000000 +0100
+++ bzr-1.10/bzrlib/tests/test_diff.py    2009-01-03 23:32:29.000000000 +0100
@@ -688,24 +688,6 @@
             r'--- olddir/oldfile.*\n\+\+\+ newdir/newfile.*\n\@\@ -1,1 \+1,1'
              ' \@\@\n-old\n\+new\n\n')

-    def test_diff_kind_change(self):
-        self.requireFeature(tests.SymlinkFeature)
-        self.build_tree_contents([('old-tree/olddir/',),
-                                  ('old-tree/olddir/oldfile', 'old\n')])
-        self.old_tree.add('olddir')
-        self.old_tree.add('olddir/oldfile', 'file-id')
-        self.build_tree(['new-tree/newdir/'])
-        os.symlink('new', 'new-tree/newdir/newfile')
-        self.new_tree.add('newdir')
-        self.new_tree.add('newdir/newfile', 'file-id')
-        self.differ.diff('file-id', 'olddir/oldfile', 'newdir/newfile')
-        self.assertContainsRe(
-            self.differ.to_file.getvalue(),
-            r'--- olddir/oldfile.*\n\+\+\+ newdir/newfile.*\n\@\@ -1,1 \+0,0'
-             ' \@\@\n-old\n\n')
-        self.assertContainsRe(self.differ.to_file.getvalue(),
-                              "=== target is 'new'\n")
-
     def test_diff_directory(self):
         self.build_tree(['new-tree/new-dir/'])
         self.new_tree.add('new-dir', 'new-dir-id')
